### ISSUE TITLE
Update py_progs to python 3

### ISSUE THREAD

**Issue by smangham** [sirocco-rt/sirocco#557]
Many of the current py_progs are not updated to python 3. They also don't have sufficient documentation to run them as intended.

**Comment by smangham** [comment]
Many also don't work as expected: 

* `balmer_decrement.py`: Doesn't specify what it's intended to test. Depends on `examples/regress/matom_balmer/balmer_test.pf`, which is outdated. After being updated, it reports the test as failed with this output: ![BalmerDecrement_balmer_test](https://user-images.githubusercontent.com/8093436/61125661-51783380-a4a2-11e9-9e2f-0674a36cf2ea.png) Probably fine, but needs validating.

* `cloudy_pl_loop.py`: Doesn't specify an example `.pf` file to compare to Cloudy. With a fresh install of the latest build of Cloudy, with `cloudy.exe` aliased to `cl`, fails on the line `subprocess.check_call("time cl input", shell=True)`. Could do with a bit more info on how to run it? @Higginbottom 

* `compare.py`: Depends on `kslio, carlo` which are not included in the distribution and not available through `pip`. Assume that @kslong has them?

**Comment by kslong** [comment]
We also need to make sure all the code is self-contained, by which I mean that everything is withing a function or class, and not in line.  This is a problem for py_docs and would be a problem for sphinx if we go this route.

**Comment by kslong** [comment]
@smangham   Do we also need to move away from py_docs and use sphinx do document our python scripts (since sphinx was written for python programs, originally)

**Comment by kslong** [comment]
These are the routines that currently fail to produce html files in py_docs.  A some of these will have failed because they need conversion to python3

```
Failed to create an html file for hydro_2_python.py
Failed to create an html file for pyfits_eval2.py
Failed to create an html file for pyfits_eval.py
Failed to create an html file for photo_xs.py
Failed to create an html file for scan_rtd_docs.py
Failed to create an html file for run_many.py
Failed to create an html file for PL_loop_comp.py
Failed to create an html file for compare_one.py
Failed to create an html file for py79_pl_loop.py
Failed to create an html file for cloudy_pl_loop.py
Failed to create an html file for compare.py
Failed to create an html file for python_pl_loop.py
```

**Comment by kslong** [comment]
To check whether Python 3 was the issue, I tried running py_docs in an environment with python2.7.  The following are still failing, so their problems are more to do with something else:

```
Failed to create an html file for hydro_2_python.py
Failed to create an html file for autogenerate_rtd_pages.py
Failed to create an html file for autogenerate_wiki_pages.py
Failed to create an html file for scan_rtd_docs.py
Failed to create an html file for run_many.py
Failed to create an html file for run_indent.py
Failed to create an html file for compare_one.py
Failed to create an html file for autogenerate_parameter_docs.py
Failed to create an html file for compare.py
```

Interestingly there are some that are appearing here that did not appear previously, e.g autogenerate_rtd_pages.py.  

My take is that we should focus on the Python3 list

**Comment by kslong** [comment]
Changes:

- run_many - There was a module date that was not used that I removed, so it works now as far as write_docs is concerned
- compare.py - this was written by ksl, and looks to be something that should be removed until he figures out what it is.  It contains refers to modules that are missing
- compare_one.py - had Python3 issues
- pyfits_eval.py - had Python3 issues and tab issues.  There are also warnings about something that is deprecated in matplotlib at present
- pyfits_eval2.py - was similar to pyfits_eval.py
- PL_loop_comp.py - had print statements and tabs that I fixed
- py79_pl_loop.py was similar
- cloudy_pl_loop.py also similar
- python_pl_loop.py

After these "fixes", the following still do not process through write_docs.py 

```
Failed to create an html file for hydro_2_python.py
Failed to create an html file for photo_xs.py
Failed to create an html file for scan_rtd_docs.py
```

Note that processing is not the same thing as working correctly, although they should work the same as before.

@smangham and @Higginbottom  this leaves only a couple of routines that do not process

hydro2_2_python is missing the pyhdf routine for a start
photo_xs.py looks just to be print and tab statements
scan_rtd_docs.py currently has a file not found error, when processed with pydoc.

**Comment by smangham** [comment]
We've got a slight issue with `write_docs.py` where running it through `UpdatePyDocs` or the symlink in the `docs/py_progs` directory means it can't actually find the modules it wants in the local directory. In addition, by calling the OS to run `pydocs`, we lose access to the virtual environment the user ran `write_docs` in. This should be fixable but will require some tweaking.

* I'll remove `scan_rtd_docs` as it was just exploratory.
* I've added installation instructions for `pyhdf` into `README.md`. Is `hydro_2_py` still necessary?
* `photo_xs.py` documents fine after running it through `2to3`.
* Attempting to fold the documentation into our main Sphinx documentation doesn't work- the docstrings are sufficiently different to RST format that Sphinx chokes on them.
